### PR TITLE
Add query groups API

### PR DESCRIPTION
**Pull request by agourlay**
This PR exposes the new query groups API and adds example snippets.

I realized there were no snippets for query so I added some.

### CODE REVIEW

File: src/builder_ext.rs [hunk 1]
```diff
@@ -537,12 +537,10 @@ impl DeleteSnapshotRequestBuilder {
 impl QueryPointGroupsBuilder {
     pub fn new(
         collection_name: impl Into<String>,
-        prefetch: impl Into<Vec<PrefetchQuery>>,
```
Comment (agourlay) [inline src/builder_ext.rs:540]: `prefetch` is not a mandatory argument

File: src/grpc_conversions/query.rs [hunk 2]
```diff
@@ -1,9 +1,28 @@
-// use crate::qdrant::{Query, query};
-//
-// impl From<Vec<f32>> for Query {
-//     fn from(v: Vec<f32>) -> Self {
-//         Query {
-//             variant: Some(query::Variant::Nearest()),
-//         }
-//     }
-// }
+use crate::qdrant::{query, Query, VectorInput};
```
Comment (agourlay) [inline src/grpc_conversions/query.rs:1]: added conversions for convenience

**Review by coszio — APPROVED** [review]
LGTM

**Review by timvisee — APPROVED** [review]
Thanks for including the example in the doc comment.

Great catch on the prefetch. We would not have been able to remove it after release.

**Pull request merged**